### PR TITLE
add filter by tags / add floor to storeys/ add show upcoming visitings

### PR DESCRIPTION
**Pull request by alikianinejad**
I have implemented several new features:

1. Users can now filter properties based on the tags that have been added.
2. Property information now includes details about the floor and the number of storeys.
3. Users can view a sorted list of upcoming property viewings directly on the map. When a filter is applied, the list will only display viewings for properties that match the selected criteria.
4. Previously, the default language was set to English (EN). If the Google Translate API wasn't configured, no descriptions would appear. I have now changed the default language to Swedish (SV), ensuring descriptions are displayed even when there is no translated version available.

### CODE REVIEW

**Comment by ewels** [comment]
This is brilliant, thanks so much @alikianinejad! I've been meaning to put in the filter for tags for ages, but never got around to it the last time I was looking for a house so it just never got done.

I wasn't able to push commits to your fork (I guess the tick box on the right of the PR was unticked), but I made a bunch of minor modifications and fixes during the merge, which you can see here: https://github.com/ewels/hemnet-commuter/compare/063a20cbd9c117e53f826cfa58dd0fa5768b743a...0e62f1d4970c0776beb5bc5edbdc8b474e72ac4a

Merged to master - thanks again!

Phil

File: app/app.js [hunk 1]
```diff
@@ -597,9 +607,75 @@ app.controller("hemnetCommuterController", ['$scope', '$compile', '$http', '$tim
...
 
+
+      // Check for `active_house_id` in URL query parameters on page load
+      const active_house_id = $location.search().active_house_id;
+
+      if (active_house_id) {
+        // If `active_house_id` exists, simulate a marker click
+        // Assuming we have a method to programmatically trigger the event
+        $scope.simulateClickOnMarker(active_house_id)
+      }
```
Comment (ewels) [inline app/app.js:618]: This sounds promising - but I can't see any code for setting the URL query parameter anywhere?

Comment (alikianinejad) [reply app/app.js:618]: Thank you, Phil, for initiating such a valuable project. I’m glad to contribute some improvements based on my experience using this project while searching for a new apartment.

Regarding the active_house_id, the original scenario was as follows: when a user clicked on a marker, the active_house_id would be updated in the query parameters (line 865). This allowed users to share or copy the URL, enabling others to see the exact house being referenced. However, I eventually removed this feature due to some usability issues. Instead, I decided to add a share button at top of the home information section, alongside the other buttons, but I haven’t had the chance to fully complete it yet.

Comment (ewels) [reply app/app.js:618]: Ok nice, understood! So it's ok to remove these lines of code then I guess?

There is an example on http://angular-ui.github.io/ui-leaflet/examples/ of updating the URL whenever the map moves for sharing, but I never got around to implementing it as I didn't think it was particularly helpful. Agree that a share button would be great though!

Comment (ewels) [reply app/app.js:618]: Note that I pushed quite a bit of new code last night, so make sure you pull in the changes to avoid merge conflicts :)

**Pull request merged**